kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/SchemaManager.java

```diff
@@ -691,6 +699,42 @@ private List<Field> getRegularSchemaFields(com.google.cloud.bigquery.Schema valu
     return result;
   }
 
+  private List<Field> validateAndRestoreFieldTypes(List<Field> proposedFields, FieldList existingFields) {
+    checkNotNull(proposedFields);
+    if (existingFields == null) {
+      return proposedFields;
+    }
+    Set<String> existingFieldNames = existingFields.stream().map(Field::getName).collect(Collectors.toSet());
```

Comment:
Can we make it case insensitive check ?
```suggestion
    Set<String> existingFieldNames = existingFields.stream().map(Field::getName).map(String::toLowercase).collect(Collectors.toSet());
```